portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -13,20 +13,89 @@
 // limitations under the License.
 
 package com.google.sps.servlets;
-
+ 
 import java.io.IOException;
+import com.google.sps.data.Book;
+import com.google.sps.data.Globals;
+import com.google.gson.Gson;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
-
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.gson.Gson;
+import java.util.ArrayList;
+import java.util.List;
+ 
+/** Servlet that handles all my received book data */
 @WebServlet("/data")
-public class DataServlet extends HttpServlet {
+public final class DataServlet extends HttpServlet {
+    
+ 
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+        List<Book> library = new ArrayList<>();
+   
+        List<Entity> results = getAllBooks(request, Globals.range);
```

Comment:
I'm not sure I completely follow the logic for Globals.range.

But wouldn't it be simpler just to pass the range as a url parameter instead of having a separate POST request to update it?